docs/src/literate-tutorials/ns_vs_diffeq.jl

```diff
@@ -105,13 +105,128 @@ nothing                    #hide
 #  \begin{bmatrix}
 #      N(\hat{v}, \hat{v}, \hat{\varphi}) \\
 #      0
-#  \end{bmatrix}
+#  \end{bmatrix}.
 # ```
 # Here $M$ is the singular block mass matrix, $K$ is the discretized Stokes operator and $N$ the nonlinear advection term, which
 # is also called trilinear form. $\hat{v}$ and $\hat{p}$ represent the time-dependent vectors of nodal values of the discretizations
 # of $v$ and $p$ respectively, while $\hat{\varphi}$ is the choice for the test function in the discretization. The hats are dropped
 # in the implementation and only stated for clarity in this section.
 #
+# ### [Derivation of the Jacobian](@id jacobian-derivation)
+# To enable the use of a wide range of solvers, it is more efficient to provide
+# information to the solver about the [sparsity pattern and values of the Jacobian](https://docs.sciml.ai/DiffEqDocs/stable/tutorials/advanced_ode_example/#stiff)
+# $J$ of $f(u, t) \equiv f(v, p, t)$ where $f(u, t)$ is just the RHS of the
+# mass matrix form given by $f(u, t) = Ku + N(u)$. By [definition](https://nl.mathworks.com/help/matlab/ref/odeset.html#d126e1203285),
+# ```math
+# J = \frac{\partial f}{\partial u} = \frac{\partial}{\partial u}(K u + N(u)) = K + \frac{\partial N}{\partial u}.
+# ```
+# It is simple to see that the Jacobian and the discretized Stokes operator $K$
+# share the same sparsity pattern since they share the same relation between
+# trial and test functions. This implies that once $K$ is assembled, its
+# sparsity values and pattern can be used to initialize the Jacobian.
+#
+# To define $\frac{\partial N}{\partial u}$, we take the directional derivative
+# of $f$ along the vector $(\delta v, \delta p)$ and
```

Comment:
I think we are mixing the derivative at the continuous weak form with the derivative of the discrete problem.